mqtt/mqtt-edgehub/src/command.rs

```diff
@@ -1,200 +1,116 @@
+            .map_err(HandleEventError::ParseClientId)?;
 
-        debug!("command handler stopped");
-    }
+        info!("received disconnection request for client {}", client_id);
 
-    async fn handle_event(&mut self, event: Event) -> Result<(), HandleDisconnectError> {
-        if let Event::Publication(publication) = event {
-            let client_id: ClientId = serde_json::from_slice(&publication.payload)
-                .map_err(HandleDisconnectError::ParseClientId)?;
-
-            info!("received disconnection request for client {}", client_id);
-
-            if let Err(e) =
-                self.broker_handle
-                    .send(Message::System(SystemEvent::ForceClientDisconnect(
-                        client_id.clone(),
-                    )))
-            {
-                return Err(HandleDisconnectError::SignalError(e));
-            }
-
-            info!(
-                "succeeded sending broker signal to disconnect client{}",
-                client_id
-            );
+        if let Err(e) = broker_handle.send(Message::System(SystemEvent::ForceClientDisconnect(
+            client_id.clone(),
+        ))) {
+            return Err(HandleEventError::DisconnectSignal(e));
         }
 
+        info!(
+            "succeeded sending broker signal to disconnect client{}",
+            client_id
+        );
         Ok(())
     }
 }
 
-async fn subscribe(
-    client: &mut mqtt3::Client<BrokerConnection>,
-    topics: &[String],
-) -> Result<(), CommandHandlerError> {
-    debug!("command handler subscribing to disconnect topic");
-    let subscriptions = topics.iter().map(|topic| proto::SubscribeTo {
-        topic_filter: topic.to_string(),
-        qos: proto::QoS::AtLeastOnce,
-    });
-
-    for subscription in subscriptions {
-        client
-            .subscribe(subscription)
-            .map_err(CommandHandlerError::SubscribeFailure)?;
+impl Command for AuthorizedIdentities {
+    fn topic(&self) -> String {
+        self.topic.clone()
     }
 
-    let mut subacks: HashSet<_> = topics.iter().map(Clone::clone).collect();
-
-    while let Some(event) = client
-        .try_next()
-        .await
-        .map_err(CommandHandlerError::PollClientFailure)?
-    {
-        if let Event::SubscriptionUpdates(subscriptions) = event {
-            for subscription in subscriptions {
-                if let SubscriptionUpdateEvent::Subscribe(sub) = subscription {
-                    subacks.remove(&sub.topic_filter);
-                }
-            }
-
-            if subacks.is_empty() {
-                debug!("command handler successfully subscribed to disconnect topic");
-                return Ok(());
-            }
+    fn handle(
+        &self,
+        broker_handle: &mut BrokerHandle,
+        publication: &ReceivedPublication,
+    ) -> Result<(), HandleEventError> {
+        let array: ServiceIdentity = serde_json::from_slice(&publication.payload)
```

Comment:
Make sure that `ServiceIdentity` struct matches the actual json payload from edge hub.